### PR TITLE
wall shit

### PR DESCRIPTION
**Pull request by Aurorablade**
Trys to pick out some stuff from tg to make some shit on walls into nice neat procs.

Hopefully making it a bit cleaner code wise for future.
I may need to do more to it and not sure if i need a changelog.

### CODE REVIEW

**Comment by Aurorablade** [comment]
anyone see anything else i need to do to this or am missing.

**Comment by Aurorablade** [comment]
rewviewableish if no one  sees anything else.

**Comment by tigercat2000** [comment]
I need to review this

**Comment by Aurorablade** [comment]
i fail patching hold on.

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -254,183 +254,194 @@
...
+		return
 
-			EB.spark_system.start()
-			to_chat(user, "<span class='notice'>You slash \the [src] with \the [EB]; the thermite ignites!</span>")
-			playsound(src, "sparks", 50, 1)
-			playsound(src, EB.usesound, 50, 1)
+	return attack_hand(user)
+
+/turf/simulated/wall/proc/try_rot(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weldingtool))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: helper check here.

File: code/game/turfs/simulated/walls.dm [hunk 2]
```diff
@@ -254,183 +254,194 @@
...
+			rotting = 0
+			return TRUE
+		else if((!is_sharp(I) && I.force >= 10) || I.force >= 20)
+			to_chat(user, "<span class='notice'>[src] crumbles away under the force of your [I.name].</span>")
+			dismantle_wall(1)
+			return TRUE
+		return FALSE
 
+/turf/simulated/wall/proc/try_thermite(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weldingtool))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: here too.

File: code/game/turfs/simulated/walls.dm [hunk 3]
```diff
@@ -254,183 +254,194 @@
...
+		if(WT.remove_fuel(0, user))
 			thermitemelt(user)
-			return
+			return TRUE
+
+	else if(istype(I, /obj/item/gun/energy/plasmacutter))
+		thermitemelt(user)
+		return TRUE
+
+	else if(istype(I, /obj/item/melee/energy/blade))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: Le ❄️

File: code/game/turfs/simulated/walls.dm [hunk 4]
```diff
@@ -254,183 +254,194 @@
...
 
-	//DECONSTRUCTION
-	if(istype(W, /obj/item/weldingtool))
+		thermitemelt(user)
+		return TRUE
+
+	return FALSE
+
+/turf/simulated/wall/proc/try_decon(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weldingtool))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: Use the helper.

File: code/game/turfs/simulated/walls.dm [hunk 5]
```diff
@@ -254,183 +254,194 @@
...
+		if(do_after(user, isdiamond ? 600 * I.toolspeed : 300 * I.toolspeed, target = src)) // Jackhammer has 0.1 toolspeed, so 60/30
+			to_chat(user, "<span class='notice'>Your [I.name] disintegrates the reinforced plating.</span>")
 			dismantle_wall()
-			visible_message("<span class='warning'>[user] disintegrates \the [src]!</span>","<span class='warning'>You hear the grinding of metal.</span>")
+			visible_message("<span class='warning'>[user] disintegrates [src]!</span>","<span class='warning'>You hear the grinding of metal.</span>")
+			return TRUE
 
-	else if(istype(W, /obj/item/melee/energy/blade))
-		var/obj/item/melee/energy/blade/EB = W
+	else if(istype(I, /obj/item/melee/energy/blade))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: more ❄️ ninja code

File: code/game/turfs/simulated/walls.dm [hunk 6]
```diff
@@ -254,183 +254,194 @@
...
+						P.setDir(9)
 					else if(user.dir == 4)
-						P.dir = 10
+						P.setDir(10)
 				else
-					P.dir = user.dir
-				P.x = src.x
-				P.y = src.y
-				P.z = src.z
-				P.loc = src
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: Good heavens almighty, why.

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 7]
```diff
@@ -33,333 +33,260 @@
...
-
-	//THERMITE related stuff. Calls src.thermitemelt() which handles melting simulated walls and the relevant effects
-	if(thermite)
-		if(iswelder(W))
-			var/obj/item/weldingtool/WT = W
-			if(WT.remove_fuel(0,user))
-				thermitemelt(user)
-				return
+/turf/simulated/wall/r_wall/proc/try_repair(obj/item/I, mob/user, params)
+	if(damage && istype(I, /obj/item/weldingtool))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls_reinforced.dm]: Helper

File: code/game/turfs/simulated/walls_reinforced.dm [hunk 8]
```diff
@@ -33,333 +33,260 @@
...
+	if(istype(I, /obj/item/pickaxe/drill/jackhammer))
 		to_chat(user, "<span class='notice'>You begin to disintegrate the wall...</span>")
 
-		if(do_after(user, 1000 * W.toolspeed, target = src)) // Jackhammer has 0.1 toolspeed, so 100
+		if(do_after(user, 1000 * I.toolspeed, target = src)) // Jackhammer has 0.1 toolspeed, so 100
 			to_chat(user, "<span class='notice'>Your sonic jackhammer disintegrates the reinforced plating.</span>")
 			dismantle_wall()
+		return TRUE
+	
+	if(istype(I, /obj/item/melee/energy/blade))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls_reinforced.dm]: 😱

**Comment by Fox-McCloud** [comment]
once conflicts are solved, I think this is ok to merge.

**Comment by Aurorablade** [comment]
wwaiting on travis.

File: code/game/turfs/simulated/walls.dm [hunk 9]
```diff
@@ -254,181 +254,168 @@
...
+	if(iswelder(I))
+		var/obj/item/weldingtool/WT = I
+		if(WT.remove_fuel(0, user))
+			to_chat(user, "<span class='notice'>You burn away the fungi with [WT].</span>")
+			playsound(src, WT.usesound, 10, 1)
+			for(var/obj/effect/overlay/wall_rot/WR in src)
+				qdel(WR)
+			rotting = 0
+			return TRUE
+		else if((!is_sharp(I) && I.force >= 10) || I.force >= 20)
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: You indented this too far; this only runs for welders.

File: code/game/turfs/simulated/walls.dm [hunk 10]
```diff
@@ -254,181 +254,168 @@
...
-			var/obj/item/melee/energy/blade/EB = W
+	// The magnetic gripper does a separate attackby, so bail from this one
+	if(istype(I, /obj/item/gripper))
+		return
 
-			EB.spark_system.start()
-			to_chat(user, "<span class='notice'>You slash \the [src] with \the [EB]; the thermite ignites!</span>")
-			playsound(src, "sparks", 50, 1)
-			playsound(src, EB.usesound, 50, 1)
+	return attack_hand(user)
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/walls.dm]: `return ..()`

**Pull request merged**